### PR TITLE
Add summary to question for textfield concatenation

### PR DESCRIPTION
**Pull request by LJBabbage**
### PR Context
Questions should be able to concatenate textfields on the summary page. This allows the element summary in question but also restricts this to just TextFields

### CODE REVIEW

File: schemas/string_interpolation/definitions.json [hunk 1]
```diff
@@ -37,6 +37,15 @@
       }
     ]
   },
+  "summary": {
```
Comment (MebinAbraham) [inline schemas/string_interpolation/definitions.json:40]: I'm not sure why this is part of this file?. This should be a property of the question, hence update the question types to have this new property?.

File: schemas/string_interpolation/definitions.json [hunk 2]
```diff
@@ -37,6 +37,15 @@
...
+    "description": "TextField string concatenations for the summary page",
+    "properties": {
+      "concatenation_type": {
+        "enum": ["NewLine", "Space"]
```
Comment (MebinAbraham) [inline schemas/string_interpolation/definitions.json:44]: Should be `Newline`?

Comment (LJBabbage) [reply schemas/string_interpolation/definitions.json:44]: Down as NewLine on the story

File: schemas/string_interpolation/definitions.json (hunk 1 above)
Comment (LJBabbage) [reply schemas/string_interpolation/definitions.json:40]: No idea why I didn't put it in the question definitions, doh

File: schemas/string_interpolation/definitions.json (hunk 2 above)
Comment (MebinAbraham) [reply schemas/string_interpolation/definitions.json]: I think we should also change `NewLine` to `Newline` as a newline character is 1 word.

File: app/validation/validator.py [hunk 3]
```diff
@@ -345,6 +345,13 @@ def _validate_questions(self, block_or_variant, numeric_answer_ranges):
                         )
                     )
 
+                if question.get("summary") and answer["type"] != "TextField":
+                    errors.append(
+                        self._error_message(
+                            f'Summary concatenation can only be used for TextFields, `{answer["id"]}` is invalid'
```
Comment (MebinAbraham) [inline app/validation/validator.py:351]: Not a blocked but this could say ``{answer["id"]}` is a `{answer["type"]}``

**Pull request merged**